lms/djangoapps/courseware/access_utils.py

```diff
@@ -168,3 +166,23 @@ def check_public_access(course, visibilities):
         return ACCESS_GRANTED
 
     return ACCESS_DENIED
+
+
+def check_data_sharing_consent(course_id):
+    """
+        Grants access if the user is do not need DataSharing consent, otherwise returns data sharing link.
+
+        Returns:
+            AccessResponse: Either ACCESS_GRANTED or DataSharingConsentRequiredAccessError
+        """
+    from openedx.features.enterprise_support.api import get_enterprise_consent_url
```

Comment:
I think we should create a followup ticket to fix this dependency issue later.

Reply:
@zamanafzal I discussed this circular import with @saleem-latif, It's ok to import locally to resolve the circular dependency in some cases.